work_plans/init_seed_data.md

```diff
@@ -0,0 +1,489 @@
+3. Mock API calls to test data aggregation logic
+4. Test CSV output validation
+
+## Usage Examples
+
+### Basic Usage (Previous Month)
+```bash
+# Generate seed data for previous month
+python scripts/generate_seed_data.py
+
+# Output:
+# Fetching players data...
+# ✓ Wrote 4,800 players to data/seed/players.csv
+# 
+# Fetching teams data...
+# ✓ Wrote 30 teams to data/seed/teams.csv
+# 
+# Fetching games data (2024-11-01 to 2024-11-30)...
+# ✓ Wrote 450 games to data/seed/games.csv
+# 
+# Fetching boxscores data for 450 games...
+# Progress: 450/450 games processed
+# ✓ Wrote 6,750 player boxscores to data/seed/boxscores.csv
+# 
+# Seed data generation complete!
+```
+
+### Custom Date Range
+```bash
+# Generate seed data for specific date range
+python scripts/generate_seed_data.py --date-from 2024-01-01 --date-to 2024-01-31
+```
+
+### Custom Output Directory
+```bash
+# Specify output directory
+python scripts/generate_seed_data.py --output-dir custom/path
+```
+
+### Via CLI
+```bash
+# Using unified CLI interface
+python main.py seed-data --output-dir data/seed
+```
+
+## API Rate Limiting Considerations
+
+The NBA API has rate limiting. Best practices:
+
+1. **Sleep between requests:** Add 0.6-second delay between API calls
+2. **Batch operations:** Process in chunks with longer delays between batches
+3. **Exponential backoff:** Retry failed requests with increasing delays
+4. **Progress indicators:** Show user that script is working (not frozen)
+
+Example implementation:
+```python
+import time
+
+def fetch_with_rate_limit(fetch_func, items, delay=0.6):
+    """Fetch data with rate limiting."""
+    results = []
+    total = len(items)
+    
+    for i, item in enumerate(items, 1):
+        result = fetch_func(item)
+        results.append(result)
+        
+        # Progress indicator
+        if i % 10 == 0 or i == total:
+            print(f"Progress: {i}/{total} items processed")
+        
+        # Rate limiting
+        if i < total:
+            time.sleep(delay)
+    
+    return results
+```
+
+## Error Handling Strategy
+
+Handle common failure scenarios gracefully:
+
+### 1. API Timeout
+```python
+try:
+    result = api_call()
+except RequestException as e:
+    print(f"API request failed: {e}")
+    print("Retrying in 5 seconds...")
+    time.sleep(5)
+    result = api_call()  # Retry once
+```
+
+### 2. Empty Results
+```python
+if df.empty:
+    print("Warning: No games found for this date range")
+    print("This may be normal if it's during the off-season")
+    return pd.DataFrame()
+```
+
+### 3. Partial Success
+```python
+# Continue processing even if some teams fail
+failed_teams = []
+for team in teams:
+    try:
+        games = fetch_games(team)
+    except Exception as e:
+        print(f"Failed to fetch games for {team}: {e}")
+        failed_teams.append(team)
+        continue
+
+if failed_teams:
+    print(f"\nWarning: Failed to fetch data for {len(failed_teams)} teams")
```

Comment:
This isn't bad, but would be helpful if it printed the team IDs or team names as well.